tests/framework/db/QueryTest.php

```diff
@@ -583,4 +583,110 @@ public function testMultipleLikeConditions()
             '%ba',
         ]));
     }
+
+    public function testGetTablesUsedInFromEmptyFrom()
+    {
+        $query = new Query;
+        $query->from([]);
+
+        $tables = $query->getTablesUsedInFrom();
+
+        $this->assertEquals([], $tables);
+    }
+
+    public function testGetTablesUsedInFromIsArrayFromAndFilledAlias()
+    {
+        $query = new Query;
+        $query->from(['u' => '{user}']);
+
+        $tables = $query->getTablesUsedInFrom();
+
+        $this->assertEquals(['{{u}}' => '{{user}}'], $tables);
+    }
+
+    public function testGetTablesUsedInFromIsArrayFromAndNotFilledAlias()
+    {
+        $query = new Query;
+        $query->from(['user']);
+
+        $tables = $query->getTablesUsedInFrom();
+
+        $this->assertEquals(['{{user}}' => '{{user}}'], $tables);
+    }
+
+    public function testGetTablesUsedInFromIsArrayFromAndEmptyAlias()
+    {
+        $query = new Query;
+        $query->from(['user AS u']);
+
+        $tables = $query->getTablesUsedInFrom();
+
+        $this->assertEquals(['{{u}}' => '{{user}}'], $tables);
+    }
+
+    public function testGetTablesUsedInFromIsStringFrom()
+    {
+        $query = new Query;
+        $query->from('user AS u, customer AS c');
+        $expected = ['{{u}}' => '{{user}}', '{{c}}' => '{{customer}}'];
+
+        $tables = $query->getTablesUsedInFrom();
+
+        $this->assertEquals($expected, $tables);
+    }
+
+    public function testGetTablesUsedInFromIsExpressionFrom()
+    {
+        $query = new Query;
+        $expression = new \yii\db\Expression('(SELECT id from user) x');
+        $query->from($expression);
+        $expected = ['{{activeusers}}' => '{{(SELECT * FROM user WHERE active = 1)}}'];
+
+        $tables = $query->getTablesUsedInFrom();
+
+        $this->assertEquals('(SELECT id from user) x', $tables);
+    }
+
+    public function testGetTablesUsedInFromIsQueryStringFrom()
+    {
+        $query = new Query;
+        $subQuery = "(SELECT * FROM `user` WHERE `active` = 1)";
+        $query->from(['activeusers' => $subQuery]);
+        $expected = ['{{activeusers}}' => '{{(SELECT * FROM user WHERE active = 1)}}'];
+
+        $tables = $query->getTablesUsedInFrom();
+
+        $this->assertEquals($expected, $tables);
+    }
+
+    public function testGetTablesUsedInFromIsQueryFrom()
+    {
+        $query = new Query;
+        $subQuery = new Query;
+        $subQuery->from('user');
+        $query->from(['x' => $subQuery]);
+        $expected = ['{{user}}' => '{{user}}'];
```

Comment:
I'm not sure what's right here.

Reply:
```php
'x' => $subQuery
```

You've specified `x` as alias explicitly and table name there doesn't make sense since it's sub-query.